src/Common/Core/Impl/Collections/EnumerableExtensions.cs

```diff
@@ -113,5 +113,11 @@ public static class EnumerableExtensions {
                 }
             }
         }
+
+        public static void ForEach<T>(this IEnumerable<T> enumerable, Action<T> action) {
```

Comment:
`List<T>` is a fixed-length read-write collection. Side effects for it is normal.
`IEnumerable<T>` is a state machine, not a collection of elements. 

`IEnumerable<T>.ForEach` instead of `foreach` sometimes means that LINQ sequence can be simplified.

For example:
```{csharp}
var list = new List<string>();
sequence1.ForEach(e => list.Add(e));
sequence2.ForEach(e => list.Add(e));
return list;
```

While it can be just:
```{csharp}
return sequence1.Concat(sequence2).ToList();
```

If `foreach` is really required in place, you can always call `sequence.AsList().ForEach()`